From 4d9be47aa0e74267e0211f566c07f6ddad566db6 Mon Sep 17 00:00:00 2001
From: Jan Wyszynski <jan.wyszynski@joyent.com>
Date: Thu, 18 Oct 2018 18:32:16 +0000
Subject: [PATCH] MANTA-4004 muskie audit should check req.metadata when
 logging delete metrics Reviewed by: Robert Bogart <robert.bogart@joyent.com>
 Approved by: Robert Bogart <robert.bogart@joyent.com>

---
 lib/audit.js | 26 ++++++++++++++------------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/lib/audit.js b/lib/audit.js
index 795bfea..fcfee03 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -299,7 +299,7 @@ function auditLogger(options) {
             latency_histogram.observe(obj.latencyToFirstByte, labels);
         }
 
-        if (op === 'DELETE') {
+        if (op === 'DELETE' && req.metadata) {
             var md = req.metadata;
             var owner = md.creator || md.owner;
 
@@ -309,17 +309,19 @@ function auditLogger(options) {
              * the deleted object was processed by the accelerated deletion
              * pipeline.
              */
-            common.checkAccountSnaplinksEnabled(req, owner,
-                function (enabled) {
-                if (md.type === 'object' && md.contentLength > 0) {
-                    var storage = md.contentLength * md.sharks.length;
-                    labels = {
-                        accelerated_gc: !enabled,
-                        owner: owner
-                    };
-                    deleted_data_counter.add(storage, labels);
-                }
-            });
+            if (owner) {
+                common.checkAccountSnaplinksEnabled(req, owner,
+                    function (enabled) {
+                    if (md.type === 'object' && md.contentLength > 0) {
+                        var storage = md.contentLength * md.sharks.length;
+                        labels = {
+                            accelerated_gc: !enabled,
+                            owner: owner
+                        };
+                        deleted_data_counter.add(storage, labels);
+                    }
+                });
+            }
         }
 
         log.info(obj, 'handled: %d', res.statusCode);
-- 
2.21.0

